dev/doxygen/pages/release_workflow.md

```diff
@@ -0,0 +1,155 @@
+major version uses exactly the same file format. And for developers it's clear
+that when changing something in the file format, the application's major version
+*must* be incremented.
+
+
+# Preprocessor Defines {#doc_release_workflow_defines}
+
+The application version is compiled into the binary with the preprocessor define
+`APP_VERSION` and the file format version with the define `FILE_FORMAT_VERSION`.
+Whether the file format is stable or not is controlled by the define
+`FILE_FORMAT_STABLE`. See following chapters for details.
+
+
+# Git Branches {#doc_release_workflow_branches}
+
+The continuous development happens on feature/bugfix branches (or pull requests),
+which are merged into the `master` branch as soon as they are finished. As new
+features sometimes also introduce file format changes, the `master` branch must
+always be considered to contain the (unstable) *next* major version of the
+application. For example if the last stable release was version "2.3.4", the
+`master` must immediately be considered as version "3.0.0-unstable".
+
+This also means that on the `master` branch we cannot provide updates for
+already released versions. So we have to provide such updates in release
+branches. For every major version we create a release branch, which is then used
+for *any* update for that major version. For example on the branch `release/2`
+we would provide the releases "2.0", "2.0.1" (bugfix) and "2.1.0" (new feature).
+
+![Git Branches](release_workflow_diagram.png)
+
+Features and bugfixes which do not affect the file format may be cherry-picked
+from `master` to release branches. This way we can publish updates for the last
+officially released application version while the next major version is still in
+development. Generally we do *not* maintain older releases than the last stable
+release because this would be too much effort. So for example as soon as the
+version "0.2.0" is released, the version "0.1.x" is no longer maintained.
+
+
+# Git Tags
+
+For every officially released version we create a Git tag on the corresponding
+release branch. The tag name contains all three numbers of the application's
+version number. Examples: "0.1.0-rc1", "0.1.0", "1.2.0", "1.2.1"
+
+
+# Warning in Unstable Versions {#doc_release_workflow_warning}
+
+Because (at least) the `master` branch is always unstable, it's very dangerous
+for end users to work productively with this version (it could break projects
+etc.!). So we should warn users in that situation.
+
+This is done by displaying warnings in the application's GUI. Their visibility
+is controlled by the preprocessor define `FILE_FORMAT_STABLE`. It's set to `0`
+(warning visible) on all branches, except on release branches where it's set to
+`1` (warning disabled). As release branches always have a stable file format, it
+should be safe to use *any* state of a release branch (even if not officially
+released).
```

Comment:
It's important that the warning is shown (and maybe acknowledged) *before* the user can open any project.

Reply:
*How* it's implemented is not part of this documentation, so you may add your comment to the PR which actually implements this warning (which does not yet exist) :)